app/lib/kafka.server.ts

```diff
@@ -0,0 +1,39 @@
+/*!
+ * Copyright © 2023 United States Government as represented by the
+ * Administrator of the National Aeronautics and Space Administration.
+ * All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { Kafka } from 'gcn-kafka'
+
+import { domain, getEnvOrDie, hostname } from './env.server'
+
+const client_id = getEnvOrDie('KAFKA_CLIENT_ID')
+const client_secret = getEnvOrDie('KAFKA_CLIENT_SECRET')
+const producer = new Kafka({
+  client_id,
+  client_secret,
+  domain,
+}).producer()
+
+process.on('SIGTERM', async () => {
+  await producer.disconnect()
+  process.exit(0)
+})
+
+export async function sendKafka(item: any, topic: string) {
+  await producer.connect()
+  await producer.send({
+    topic,
+    messages: [
+      {
+        key: 'message',
+        value: JSON.stringify(item),
+      },
+    ],
+  })
+  if (hostname === 'localhost') {
+    await producer.disconnect()
+  }
```

Comment:
What is this for?

```suggestion
```

Reply:
I left a comment in a previous thread, but there is no SIGTERM in the sandbox mode. If you don't disconnect eventually, the lambda hangs and throws an error

Reply:
Alright. Please submit a bug report for Architect and explain why this is here with a FIXME comment.

And please do close the connection at exit.

Reply:
It may not be architect specifically. For local testing we are triggering this with the invoker plugin. I'm testing their end function to see if maybe that can add an emit without totally also killing the main app too

Reply:
Nevermind, I am looking over the [sandbox invoke](https://github.com/architect/sandbox/blob/main/src/invoke-lambda/index.js) code and think its probably because of how this runs